source/parameters.cc

```diff
@@ -1383,7 +1383,51 @@ namespace WorldBuilder
           }
         else if (type == "object")
           {
-            collapse += "/properties";
+            // normal objects just have properties, but some have oneOf
+            if (Pointer((base_path + "/oneOf").c_str()).Get(declarations) != nullptr)
+              {
+                // it has a structure with oneOf. Find out which of the entries is needed.
+                // This means we have to take a sneak peak to figure out how to get to the
+                // next value.
+                size_t size = Pointer((base_path + "/oneOf").c_str()).Get(declarations)->Size();
+#ifdef debug
+                bool found = false;
+#endif
```

Comment:
This has nothing to do with the patch itself, but just as a comment: I assume that you are putting the `#ifdef` around this statement because you would otherwise get a warning about the `found` variable only ever being written to, but never read, in release mode? If so, you can avoid the warning if you put
```
  (void)found;
```
in front of the `WBAssert`. This is interpreted as a "read" of the variable, but you are just discarding the value so read.

Reply:
interesting, I will try that!

Reply:
on my laptop those if statement do not seem to matter, but probably for one of the tester configurations it does. I will look into it in a different pull request.